lib/rubygems/version.rb

```diff
@@ -381,10 +389,26 @@ def <=>(other)
   end
 
   def canonical_segments
-    @canonical_segments ||=
-      _split_segments.map! do |segments|
```

Comment:
This was previously mutated. Is it ok to no longer mutate it?

Reply:
There was an issue with one gem relying being able to mutate this array. See https://github.com/rubygems/rubygems/pull/4247. I think that issue shows that it's best to keep this frozen to avoid those kind of issues, but also shows this could potentially break something.

I think it's ok after but just wanted to mention it!

Reply:
Wow. That's a subtle interaction. It reinforces the choice to make these frozen. Thanks for replying @deivid-rodriguez.